blueocean-dashboard/src/main/js/components/Branches.jsx

```diff
@@ -32,7 +32,8 @@ export default class Branches extends Component {
             router.push(location);
         };
 
-        const { msg } = (branch.changeSet && branch.changeSet.length > 0) ? (branch.changeSet[0] || {}) : {};
+        const { msg } = (latestRun.changeSet && latestRun.changeSet.length > 0) ? (latestRun.changeSet[latestRun.changeSet.length - 1] || {}) : {};
+        
```

Comment:
This would be a lot more readable split into two less-clever lines I feel, but maybe that's just me.